controllers/topology/desired_state.go

```diff
@@ -170,6 +182,92 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 }
 
+// computeMachineDeployment computes the desired state for a MachineDeploymentTopology.
+// The generated machineDeployment object is calculated using the values from the machineDeploymentTopology and
+// the machineDeployment class.
+func computeMachineDeployment(class *clusterTopologyClass, current *clusterTopologyState, machineDeploymentTopology clusterv1.MachineDeploymentTopology) (machineDeploymentTopologyState, error) {
+	desiredMachineDeployment := machineDeploymentTopologyState{}
+
+	className := machineDeploymentTopology.Class
+	machineDeploymentTopologyClass, ok := class.machineDeployments[className]
+	if !ok {
+		return desiredMachineDeployment, errors.Errorf("class name %s not found in clusterClass %s", className, class.clusterClass.Name)
+	}
+
+	currentMachineDeployment := current.machineDeployments[machineDeploymentTopology.Name]
+	var currentBootstrapTemplateRef *corev1.ObjectReference
+	if currentMachineDeployment.bootstrapTemplate != nil {
+		currentBootstrapTemplateRef = currentMachineDeployment.object.Spec.Template.Spec.Bootstrap.ConfigRef
+	}
+	desiredMachineDeployment.bootstrapTemplate = templateToTemplate(templateToInput{
+		template:              machineDeploymentTopologyClass.bootstrapTemplate,
+		templateClonedFromRef: objToRef(machineDeploymentTopologyClass.bootstrapTemplate),
+		cluster:               current.cluster,
+		namePrefix:            fmt.Sprintf("%s-%s-bootstrap-", current.cluster.Name, machineDeploymentTopology.Name),
+		currentObjectRef:      currentBootstrapTemplateRef,
+		labels:                mergeMap(machineDeploymentTopology.Metadata.Labels, machineDeploymentTopologyClass.metadata.Labels),
+		annotations:           mergeMap(machineDeploymentTopology.Metadata.Annotations, machineDeploymentTopologyClass.metadata.Annotations),
+	})
+
+	var currentInfraMachineTemplateRef *corev1.ObjectReference
+	if currentMachineDeployment.infrastructureMachineTemplate != nil {
+		currentInfraMachineTemplateRef = &currentMachineDeployment.object.Spec.Template.Spec.InfrastructureRef
+	}
+	desiredMachineDeployment.infrastructureMachineTemplate = templateToTemplate(templateToInput{
+		template:              machineDeploymentTopologyClass.infrastructureMachineTemplate,
+		templateClonedFromRef: objToRef(machineDeploymentTopologyClass.infrastructureMachineTemplate),
+		cluster:               current.cluster,
+		namePrefix:            fmt.Sprintf("%s-%s-infra-", current.cluster.Name, machineDeploymentTopology.Name),
+		currentObjectRef:      currentInfraMachineTemplateRef,
+		labels:                mergeMap(machineDeploymentTopology.Metadata.Labels, machineDeploymentTopologyClass.metadata.Labels),
+		annotations:           mergeMap(machineDeploymentTopology.Metadata.Annotations, machineDeploymentTopologyClass.metadata.Annotations),
+	})
+
+	gv := clusterv1.GroupVersion
+	desiredMachineDeploymentObj := &clusterv1.MachineDeployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       gv.WithKind("MachineDeployment").Kind,
```

Comment:
```suggestion
			Kind:       "MachineDeployment",
```
nit, Seems to be identical to just setting the kind directly. (in this case gv could be inlined in the next line to avoid the additional local var).

Reply:
Saw a similar pattern when creating `Machine` objects, tried to follow the same.